tests/frequentist_stats/test_linear_models.py

```diff
@@ -0,0 +1,510 @@
+    for bad_alphas in [-1, 0, 1, 2]:
+        with pytest.raises(ValueError, match=r"alpha must be in \(0,1\)"):
+            mafslm.stringify_alpha(bad_alphas)
+
+    alpha = 0.1
+    low, high = mafslm.stringify_alpha(alpha)
+    assert low == "0.050"
+    assert high == "0.950"
+
+    alpha = 0.05
+    low, high = mafslm.stringify_alpha(alpha)
+    assert low == "0.025"
+    assert high == "0.975"
+
+    alpha = 0.01
+    low, high = mafslm.stringify_alpha(alpha)
+    assert low == "0.005"
+    assert high == "0.995"
+
+
+def test_summarize_one_branch():
+    test_data = pd.Series(range(100))
+    alphas = [0.05]
+    actuals = mafslm.summarize_one_branch(test_data, alphas)
+
+    mean = 49.5
+    low, high = 43.74349, 55.25650
+
+    assert np.isclose(actuals["0.5"], mean)
+    assert np.isclose(actuals["mean"], mean)
+    assert np.isclose(actuals["0.025"], low)
+    assert np.isclose(actuals["0.975"], high)
+
+    index_values = actuals.index.values
+    index_values.sort()
+    assert list(index_values) == ["0.025", "0.5", "0.975", "mean"]
+
+
+def test_summarize_univariate():
+    one_branch_data = pd.Series(range(100))
+    test_data = pd.concat([one_branch_data, one_branch_data])
+    branches = pd.Series([*(["control"] * 100), *(["treatment"] * 100)])
+    branch_list = ["control", "treatment"]
+
+    result = mafslm.summarize_univariate(test_data, branches, branch_list, [0.05])
+
+    assert sorted(result.keys()) == branch_list
+
+    # validate against theoretical values
+    mean = 49.5
+    low, high = 43.74349, 55.25650
+    for branch in branch_list:
+        assert np.isclose(result[branch]["mean"], mean)
+        assert np.isclose(result[branch]["0.5"], mean)
+        assert np.isclose(result[branch]["0.025"], low)
+        assert np.isclose(result[branch]["0.975"], high)
+
+        index_values = result[branch].index.values
+        index_values.sort()
+        assert list(index_values) == ["0.025", "0.5", "0.975", "mean"]
+
+    # cross-validate against existing bootstrap implementation
+    bootstrap_result = mafsb.bootstrap_one_branch(one_branch_data)
+    for branch in branch_list:
+        assert np.isclose(result[branch]["mean"], bootstrap_result["mean"], atol=0.5)
+        assert np.isclose(result[branch]["0.5"], bootstrap_result["0.5"], atol=0.5)
+        assert np.isclose(result[branch]["0.025"], bootstrap_result["0.025"], atol=0.5)
+        assert np.isclose(result[branch]["0.975"], bootstrap_result["0.975"], atol=0.5)
+
+
+def test__make_formula():
+    expected = "search_count ~ C(branch, Treatment(reference='control'))"
+    actual = mafslm._make_formula("search_count", "control")
+
+    assert expected == actual
+
+    expected = "days_of_use ~ C(branch, Treatment(reference='treatment-a'))"
+    actual = mafslm._make_formula("days_of_use", "treatment-a")
+
+    assert expected == actual
+
+    expected = (
+        "active_hours ~ C(branch, Treatment(reference='treatment-a'))"
+        " + active_hours_pre"
```

Comment:
nit picking but it'd be appropriate to ignore the line length rule here imo

Reply:
TIL you could do that.